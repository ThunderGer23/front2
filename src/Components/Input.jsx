

export const Input = ({Tipo,Holder}) =>{
    return (
        <>
        <div className="input-group mb-4">
            <span className="input-group-text" id="basic-addon1">
                <i className="fa-regular fa-user" style={{color:'white'}}></i>
            </span>
        </div>
        <input type={Tipo} className="form-control" placeholder={Holder} aria-label="Username" aria-describedby="basic-addon1"/>
        </>
    )
}

