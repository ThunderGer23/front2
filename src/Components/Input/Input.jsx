import PropTypes from 'prop-types'

export const Input = ({Tipo,Holder}) =>{
    return (
        <>
        <div role='div' className="input-group mb-4">
            <span role='span' className="input-group-text" id="basic-addon1">
                <i role='i' className="fa-regular fa-user" style={{color:'white'}}></i>
            </span>
        </div>
        <input role='input' type={Tipo} className="form-control" placeholder={Holder} aria-label="Username" aria-describedby="basic-addon1"/>
        </>
    )
}

Input.propTypes = {
    Tipo: PropTypes.string.isRequired,
    Holder: PropTypes.string.isRequired
}