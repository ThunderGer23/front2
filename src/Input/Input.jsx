export const Input = () =>{
    return (
        <>
        <div className="input-group mb-4">
            <span className="input-group-text" id="basic-addon1">
                <i className="fa-regular fa-user" style={{color:'white'}}></i>
            </span>
        </div>
        <input type="text" className="form-control" placeholder="usuario:" aria-label="Username" aria-describedby="basic-addon1"/>
        </>
    )
}