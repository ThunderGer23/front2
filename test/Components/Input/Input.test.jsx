import { render, screen } from "@testing-library/react";
import { Input } from "../../../src/Components/Input/Input";

describe('Pruebas en el componente Input', () => { 
    const Tipo='email'
    const Holder='email:'

    test('Esta prueba es para hacer match con el snapshot', () => { 
        const {container} = render(<Input Tipo={Tipo} Holder={Holder}/>)//Renderiza el conponente Input
        expect(container).toMatchSnapshot()//Toma una foto
     })

    test('Esta prueba valida que el div exista y pertenezca a la clase correcta', () => { 
        render(<Input Tipo={Tipo} Holder={Holder}/>)
        expect(screen.findByRole('div')).toBeTruthy()
        const {className} = screen.getByRole('div')//Se salva el tipo de la clase del div en className
        expect(className).toBe('input-group mb-4')
     })

    test('Esta prueba valida que el span exista y pertenezca a la clase correcta', () => { 
        render(<Input Tipo={Tipo} Holder={Holder}/>)
        expect(screen.findByRole('span')).toBeTruthy()
        const {className} = screen.getByRole('span')//Se salva el tipo de clase del span en className
        expect(className).toBe('input-group-text')
     })

    test('Esta prueba valida que el i exista y pertenezca a la clase correcta', () => { 
        render(<Input Tipo={Tipo} Holder={Holder}/>)
        expect(screen.findByRole('i')).toBeTruthy()
        const {className} = screen.getByRole('i')
        expect(className).toBe('fa-regular fa-user')
     })

    test('Esta prueba valida que el input exista y pertenezca a la clase correcta', () => { 
        render(<Input Tipo={Tipo} Holder={Holder}/>)
        expect(screen.findByRole('input')).toBeTruthy()
        const {className} = screen.getByRole('input')
        expect(className).toBe('form-control')
     })
 })