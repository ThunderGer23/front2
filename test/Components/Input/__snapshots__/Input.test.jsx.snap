// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Input Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <div
    class="input-group mb-4"
    role="div"
  >
    <span
      class="input-group-text"
      id="basic-addon1"
      role="span"
    >
      <i
        class="fa-regular fa-user"
        role="i"
        style="color: white;"
      />
    </span>
  </div>
  <input
    aria-describedby="basic-addon1"
    aria-label="Username"
    class="form-control"
    placeholder="email:"
    role="input"
    type="email"
  />
</div>
`;
